reactnative/RNPlayground/app/screens/PlaygroundScreen.tsx

```diff
@@ -1,76 +1,108 @@
-import React, { FC } from "react"
+import * as React from "react"
 import { observer } from "mobx-react-lite"
-import { StackScreenProps } from "@react-navigation/stack"
-import { AppStackScreenProps } from "../navigators"
-import { Image, ImageStyle, TextStyle, View, ViewStyle } from "react-native"
-import {
-  Text, Screen
-} from "../components"
-import { isRTL } from "../i18n"
+import { BaseScreenProps } from "../navigators"
+import { TextStyle, View, ViewStyle } from "react-native"
+import { Text, Button } from "../components"
 import { colors, spacing } from "../theme"
 import { useSafeAreaInsetsStyle } from "../utils/useSafeAreaInsetsStyle"
+import { useFocusEffect } from "@react-navigation/core"
+import { useStores } from "../models"
 
+interface PlaygroundScreenProps extends BaseScreenProps<"Playground"> {
+}
 
-export const PlaygroundScreen: FC<StackScreenProps<AppStackScreenProps, "Playground">> = observer(function PlaygroundScreen() {
-    const $bottomContainerInsets = useSafeAreaInsetsStyle(["bottom"])
+export const PlaygroundScreen: React.FC<PlaygroundScreenProps> = observer((props) => {
+  const $bottomContainerInsets = useSafeAreaInsetsStyle(["bottom"])
+  const { playgroundStore } = useStores()
 
-    return (
-       <View style={$container}>
-         <View style={$topContainer}>
-           <Text
-             testID="welcome-heading"
-             style={$welcomeHeading}
-             tx="playgroundScreen.readyForLaunch"
-             preset="heading"
-           />
-           <Text tx="playgroundScreen.postscript" preset="subheading" />
-         </View>
+  useFocusEffect(
+    React.useCallback(() => {
+      playgroundStore.startCounter()
+      return () => playgroundStore.stopCounter()
+    }, [playgroundStore])
+  )
 
-         <View style={[$bottomContainer, $bottomContainerInsets]}>
-           <Text tx="playgroundScreen.exciting" size="md" />
-         </View>
-       </View>
-     )
- })
+  return <View style={$container}>
+    {
+      StateComponent(
+        $bottomContainerInsets,
+        playgroundStore.uiState,
+        () => playgroundStore.resetCounter()
+      )
+    }
+  </View>
+})
 
- const $container: ViewStyle = {
-   flex: 1,
-   backgroundColor: colors.background,
- }
+function StateComponent(insets, uiState, onResetClicked) {
+  const { loading, error, content } = uiState
+  if (loading) {
```

Comment:
Can we make this a Switch Case operator?

Reply:
I'm not sure how to write a switch like that, these are separate properties of the `uiState` object where each can be either there or `undefined`